client/cmd.go

```diff
@@ -147,6 +149,24 @@ func ReadPersistentCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Cont
 		}
 	}
 
+	if clientCtx.GRPCClient == nil || flagSet.Changed(flags.FlagGRPC) {
+		grpcURI, _ := flagSet.GetString(flags.FlagGRPC)
+		if grpcURI != "" {
+			var dialOpts []grpc.DialOption
+
+			useInsecure, _ := flagSet.GetBool(flags.FlagGRPCInsecure)
```

Comment:
Turns out there's no scenario where insecure is not required. gRPC always requires some sort of client credentials. I'm not sure whether it will still use TLS if we provide insecure credentials, I'm assuming not which seems really weird. Would appreciate input if others have context

Reply:
It may be possible to parse the URL here and choose the dial option without the extra flag?

Example:
```golang
func NewGrpcConnection(endpoint string) (*grpc.ClientConn, error) {
	grpcUrl, err := url.Parse(endpoint)
	if err != nil {
		return nil, err
	}

	var secureOpt grpc.DialOption
	switch grpcUrl.Scheme {
	case "http":
		secureOpt = grpc.WithInsecure()
	case "https":
		creds := credentials.NewTLS(&tls.Config{})
		secureOpt = grpc.WithTransportCredentials(creds)
	default:
		return nil, fmt.Errorf("unknown grpc url scheme: %s", grpcUrl.Scheme)
	}

	grpcConn, err := grpc.Dial(grpcUrl.Host, secureOpt)
	if err != nil {
		return nil, err
	}

	return grpcConn, nil
}
```

The downside of not having an explicit flag is connecting to an `http` endpoint by mistake wouldn't error if the endpoint accepts both http & https connections.

Reply:
Thank you @nddeluca! Will give this a try

Reply:
Figured out a way to default to using TLS credentials if insecure isn't set and looks like it works. Thanks for your hint @nddeluca 🙏